test/e2e/scrapeconfig_test.go

```diff
@@ -540,6 +543,62 @@ func runScrapeConfigCRDValidation(t *testing.T, testCases []scrapeCRDTestCase) {
 	}
 }
 
+var HTTPSDTestCases = []scrapeCRDTestCase{
+	{
+		name: "Invalid URL",
+		scrapeConfigSpec: monitoringv1alpha1.ScrapeConfigSpec{
+			HTTPSDConfigs: []monitoringv1alpha1.HTTPSDConfig{
+				{
+					URL: "valid-server",
+				},
+			},
+		},
+		expectedError: true,
+	},
+	{
+		name: "Invalid empty URL",
+		scrapeConfigSpec: monitoringv1alpha1.ScrapeConfigSpec{
+			HTTPSDConfigs: []monitoringv1alpha1.HTTPSDConfig{
+				{
+					URL: "",
+				},
+			},
+		},
+		expectedError: true,
+	},
+	{
+		name: "Invalid absent URL",
+		scrapeConfigSpec: monitoringv1alpha1.ScrapeConfigSpec{
+			HTTPSDConfigs: []monitoringv1alpha1.HTTPSDConfig{
+				{},
+			},
+		},
+		expectedError: true,
+	},
+	{
+		name: "Valid URL with http scheme",
+		scrapeConfigSpec: monitoringv1alpha1.ScrapeConfigSpec{
+			HTTPSDConfigs: []monitoringv1alpha1.HTTPSDConfig{
+				{
+					URL: "http://valid-url",
```

Comment:
(nit) a good habit is to use [reserved top-level domains](https://www.rfc-editor.org/rfc/rfc2606.html#page-2) for tests and documentation (the existing code base doesn't apply this recommendation though!). 
```suggestion
					URL: "http://valid.test",
```

Reply:
Hmm... maybe we can just log an issue for that, it would make a good first-issue IMO.